gettsim/benefits/unterhaltsvorschuss.py

```diff
@@ -1,41 +1,41 @@
-def uhv(tax_unit, params):
+import numpy as np
+
+
+def uhv(tax_unit, params, kindergeld_params):
     """
-    Since 2017, the receipt of this
-    UHV has been extended substantially and needs to be taken into account, since it's
-    dominant to other transfers, i.e. single parents 'have to' apply for it.
+    Advance on Alimony Payment / Unterhaltsvorschuss (UHV)
+
+    Single Parents get alimony payments for themselves and for their
+    child from the ex partner. If the ex partner is not able to pay the child
+    alimony, the government pays the child alimony to the mother (or the father, if
+    he has the kids)
+
+    The amount is specified in §1612a BGB and, ultimately, in Mindesunterhaltsverordnung.
 
     returns:
     tax_unit: Updated DataFrame including uhv
     """
-    if params["year"] >= 2017:
-        return uhv_since_2017(tax_unit, params)
-    else:
-        tax_unit["uhv"] = 0
-        return tax_unit
 
+    return params["uhv_calc"](tax_unit, params, kindergeld_params)
 
-def uhv_since_2017(tax_unit, params):
-    """ Advanced Alimony Payment / Unterhaltsvorschuss (UHV)
 
-        In Germany, Single Parents get alimony payments for themselves and for their
-        child from the ex partner. If the ex partner is not able to pay the child
-        alimony, the government pays the child alimony to the mother (or the father, if
-        he has the kids)
+def uhv_since_07_2017(tax_unit, params, kindergeld_params):
+    """ UHV ruling since 07/2017. Before 2017, basically the same, but
+        eligibility was more restrictive.
 
         returns:
             tax_unit: Updated DataFrame including uhv
         """
-    # Benefit amount depends on parameters M (rent) and Y (income) (§19 WoGG)
-    # Calculate them on the level of the tax unit
-
+    # Benefit amount depends on the tax allowance for children ("sächliches Existenzminimum")
+    # and the child benefit for the first child.
     tax_unit["uhv"] = 0
     # Amounts depend on age
-    tax_unit.loc[tax_unit["age"].between(0, 5) & tax_unit["alleinerz"], "uhv"] = params[
-        "uhv5"
-    ]
+    tax_unit.loc[(tax_unit["age"] <= 6) & tax_unit["alleinerz"], "uhv"] = (
```

Comment:
This is
> bis zur Vollendung des sechsten Lebensjahrs

I think usually the age variable will be the number of completed years. So for someone 6 years and 6 month old this variable will be either 6 or 6.5. I think in both cases they will be in the second age group.
 
is it better to write it as <6?